library/core/src/ptr/const_ptr.rs

```diff
@@ -101,6 +99,53 @@ impl<T: ?Sized> *const T {
         if self.is_null() { None } else { unsafe { Some(&*self) } }
     }
 
+    /// Returns `None` if the pointer is null, or else returns a shared reference to
+    /// the value wrapped in `Some`.
```

Comment:
This should probably mention "uninitialized" somewhere. It could point out explicitly that the key difference to `as_ref`. is that the pointed-to data does *not* have to be initialized.

Reply:
May I add something like "In comparison to [`as_ref`](link to as_ref), this does not require, that the value has to be initialized."?

Reply:
I'd probably say "in contrast to", and the last "," doesn't belong there I think, but yes. :)

Reply:
Alright. Well, the last "," is German grammar I think 😄

Reply:
I have also added a link from `as_ref` to `as_uninit_ref` to quickly jump back.